lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -756,6 +761,77 @@ void TypeLoweringVisitor::visitStmt(ConnectOp op) {
   opsToRemove.push_back(op);
 }
 
+void TypeLoweringVisitor::recursivePartialConnect(Value a, FIRRTLType aType,
+                                                  Value b, FIRRTLType bType,
+                                                  Twine suffix, bool aFlip) {
+  TypeSwitch<FIRRTLType>(aType)
+      .Case<BundleType>([&](auto aType) {
+        auto bBundle = bType.dyn_cast_or_null<BundleType>();
+        if (!bBundle)
+          return;
+        for (auto aElt : aType.getElements()) {
+          auto aField = aElt.name.getValue();
+          auto bElt = bBundle.getElement(aField);
+          if (bElt) {
+            if (suffix.isTriviallyEmpty())
+              recursivePartialConnect(a, aElt.type, b, bElt.getValue().type,
+                                      aField, aFlip);
+            else
+              recursivePartialConnect(a, aElt.type, b, bElt.getValue().type,
+                                      suffix + "_" + aField, aFlip);
```

Comment:
I'm trying to buy a bit more horizontal space here.  It might be worth it to create a String instead of using a Twine like I did here: https://github.com/llvm/circt/pull/909#discussion_r615287133

```c++
if (!bElt)
  continue;
auto fieldSuffix = aField;
if (!suffix.isTriviallyEmpty())
  fieldSuffix = (suffix + "_" + a_field).str();
recursivePartialConnect(a, aElt.type, b, bElt.getValue().type,
                        fieldSuffix, aFlip);
 ```

Reply:
I was trying _very_ hard to use `Twine` here until I actually needed the string. I do think the usage here is safe. However, this is pretty unclear because a safe usage seems to require two kind paths.

I'll switch to your solution here and remove the two code paths. I'll also remove the two code paths in later `FVectorType` case. That should save both horizontal and vertical space as well as improve clarity.